### PR TITLE
s4879083 3d-unet

### PR DESCRIPTION
**Pull request by Han1zen**

### CODE REVIEW

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>No design and implementation. </td><td>-2</td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Shown dice score in graph not consistant with reported score. Train loss > Validation loss looks incorrect. DL refer to files that’s not inlcude (ex:train_list.txt)</td><td>-3</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting missing. No doc strings.</td><td>-1</td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Normal. UNet3D  (Normal Difficulty)</td><td>-5</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. </td><td>-1</td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td>-2</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. Some markup errors.</td><td>-2</td></tr>
	<tr><td></td><td>Model/technical explanation :Acceptable. Model details minimal. </td><td>-1</td></tr>
	<tr><td></td><td>Description and Comments :Acceptable. Usage instructions basic/has gaps.</td><td>-1</td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF submitted.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Pull Request has problems: Late submission.</td><td>-2</td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is missing. Empty</td><td>-2</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-23</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by celinsze** [comment]
**This is an initial inspection, no action is required at this point**

Recognition problem:
Problem solving: 
•	Provided average Dice score of above 0.7 but with training and validation loss > 0.6 after final epoch.
•	Dice score for individual label not provided for evaluation.
•	No example output provided.
Algorithm implementation: 
•	Algorithm implemented in PyTorch for the UNet3D model and functions.
Design: 
•	Loss function defined but not used for testing (predict.py)
Commenting: 
•	Some comments throughout the code but missing comments for model
Difficulty: 3D UNet – Normal

**Comment by shakes76** [comment]
Additional -1 for late submission of PR, but all work done on time.
